From 87622041abb1387bc8580721d23ec79a03e5249c Mon Sep 17 00:00:00 2001
From: CTCaer <ctcaer@gmail.com>
Date: Sun, 6 Mar 2022 04:26:06 +0000
Subject: [PATCH 03/20] codecs: nvv4l2: avoid probing

NVV4L2 does not support probing for getting stream metadata.
So disallow that.
---
 libavcodec/nvv4l2_dec.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libavcodec/nvv4l2_dec.c b/libavcodec/nvv4l2_dec.c
index ab7f0ee45e..edceebddf8 100644
--- a/libavcodec/nvv4l2_dec.c
+++ b/libavcodec/nvv4l2_dec.c
@@ -1085,7 +1085,8 @@ nvv4l2dec_decode(AVCodecContext *avctx, void *data, int *got_frame,
         .decode         = nvv4l2dec_decode,                                           \
         .flush          = nvv4l2dec_flush,                                            \
         .priv_class     = &nvv4l2_##NAME##_dec_class,                                 \
-        .capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_HARDWARE,                 \
+        .capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_HARDWARE |                \
+                          AV_CODEC_CAP_AVOID_PROBING,                                 \
         .bsfs           = BSFS,                                                       \
         .wrapper_name   = "nvv4l2",                                                   \
         .pix_fmts       =(const enum AVPixelFormat[]){ AV_PIX_FMT_YUV420P,            \
-- 
2.25.1

